hypnotoad/core/mesh.py

```diff
@@ -2517,6 +2517,15 @@ def writeGridfile(self, filename):
         from boututils.datafile import DataFile
 
         with DataFile(filename, create=True, format="NETCDF4") as f:
+            # Save unique ID for grid file
+            import uuid
+
+            id_str = str(uuid.uuid1())
+            id_numpy_array = numpy.array(list(id_str), dtype="S1")
```

Comment:
Is this to match what BOUT++ expects, a fixed size array of characters?

Reply:
Yes, exactly. I copied the pattern from `test-restart-io/runtest` in https://github.com/boutproject/BOUT-dev/pull/2192, so I think it's what's needed for BOUT++ to read the string correctly. Possibly a file attribute would be better?? BOUT++ supports them (at the moment, what's the status in `OptionsNetCDF`?), but `boututils.datafile.DataFile` doesn't at the moment and I don't think `Mesh::get()` does, so this seems like the best way.

Reply:
I think a variable is fine. `OptionsNetCDF` does support attributes, and I think it probably handles string variables/attributes better than Datafile, but I've not tested it personally (though there are tests! I'm just not sure what string-type it actually saves them as)

Reply:
Just found out / remembered that `mesh->get()` can get strings from netCDF 'global attributes'. I think we should write them that way for consistency with IDL hypnotoad, and so we can read them already with `mesh->get()` (see #68).

Reply:
Now implemented saving as file attribute instead of variable, but needs an update to `boututils` https://github.com/boutproject/boututils/pull/31, so converting to draft for now.